webapp/src/main/webapp/iaf/gui/views/TestPipeline.html

```diff
@@ -19,6 +19,15 @@ <h4>Test a PipeLine</h4>
                                 </select>
                             </div>
                         </div>
+						<div class="row form-group test-pipeline-session-keys" ng-repeat="order in sessionKeyIndices">
+							<label class="col-sm-3 control-label label-height-30"><span ng-if="$index==0">Session Key</span> </label>
+							<div class="col-sm-4">
+								<input class="form-control" ng-blur="updateSessionKeys(form.sessionKey[order], form.sessionKeyValue[order], order)" name="sessionKey" placeholder="Name" type="text" ng-model="form.sessionKey[order]"/>
+							</div>
+							<div class="col-sm-5">
+								<input class="form-control" ng-blur="updateSessionKeys(form.sessionKey[order], form.sessionKeyValue[order], order)" name="sessionKeyValue" placeholder="Value" type="text" ng-model="form.sessionKeyValue[order]"/>
```

Comment:
I think it might be easier/cleaner to use `form.sessionkeys[$index].name` and `form.sessionkeys[$index].value`
That you avoid all these for loops and the key/value pair will always be a pair.
You can put `ng-init="form.sessionkeys[$index] = {};"` behind ng-repeat="order in sessionKeyIndices" to automatically initialize a new row.